services/brig/src/Brig/API/User.hs

```diff
@@ -316,20 +316,22 @@ createUser new@NewUser {..} = do
 -- | docs/reference/user/registration.md {#RefRestrictRegistration}.
 checkRestrictedUserCreation :: NewUser -> ExceptT CreateUserError AppIO ()
 checkRestrictedUserCreation new = do
-  let memail = newUserEmail new
-      mphone = newUserPhone new
-      msso = newUserSSOId new
-      mteam = newUserTeam new
+  let nTeam = newUserTeam new
+      nExpires = newUserExpiresIn new
+
   restrictPlease <- lift . asks $ fromMaybe False . setRestrictUserCreation . view settings
   when
     ( restrictPlease
-        && or [isJust memail, isJust mphone]
-        && and [isNothing msso, isNotTeamMember mteam]
+           && not (or [isTeamMember nTeam, isEphemeral nExpires])
```

Comment:
```suggestion
           && not (isTeamMember nTeam)
           && not (isEphemeral nExpires)
```

Looks easier to reason about like this to me (because I am very challenged in boolean algebra). Feel free to ignore.